<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2016-01-04 Mon 16:56 -->
<meta  http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta  name="viewport" content="width=device-width, initial-scale=1" />
<title></title>
<meta  name="generator" content="Org-mode" />
<meta  name="author" content="Samuel BARRETO" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: visible;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline;}
  pre.src-sh:before    { content: 'sh'; }
  pre.src-bash:before  { content: 'sh'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-R:before     { content: 'R'; }
  pre.src-perl:before  { content: 'Perl'; }
  pre.src-java:before  { content: 'Java'; }
  pre.src-sql:before   { content: 'SQL'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  /*]]>*/-->
</style>
<script type="text/javascript">
/*
@licstart  The following is the entire license notice for the
JavaScript code in this tag.

Copyright (C) 2012-2013 Free Software Foundation, Inc.

The JavaScript code in this tag is free software: you can
redistribute it and/or modify it under the terms of the GNU
General Public License (GNU GPL) as published by the Free Software
Foundation, either version 3 of the License, or (at your option)
any later version.  The code is distributed WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE.  See the GNU GPL for more details.

As additional permission under GNU GPL version 3 section 7, you
may distribute non-source (e.g., minimized or compacted) forms of
that code without the copy of the GNU GPL normally required by
section 4, provided you include this license notice and a URL
through which recipients can access the Corresponding Source.


@licend  The above is the entire license notice
for the JavaScript code in this tag.
*/
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>
<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        displayAlign: "center",
        displayIndent: "0em",

        "HTML-CSS": { scale: 100,
                        linebreaks: { automatic: "false" },
                        webFont: "TeX"
                       },
        SVG: {scale: 100,
              linebreaks: { automatic: "false" },
              font: "TeX"},
        NativeMML: {scale: 100},
        TeX: { equationNumbers: {autoNumber: "AMS"},
               MultLineWidth: "85%",
               TagSide: "right",
               TagIndent: ".8em"
             }
});
</script>
<script type="text/javascript"
        src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
</head>
<body>
<div id="content">
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#orgheadline1">1. Structure</a></li>
<li><a href="#orgheadline9">2. Commentaires</a>
<ul>
<li><a href="#orgheadline2">2.1. <span class="timestamp-wrapper"><span class="timestamp">[2015-11-09 Mon]</span></span></a></li>
<li><a href="#orgheadline3">2.2. <span class="timestamp-wrapper"><span class="timestamp">[2015-11-16 Mon]</span></span></a></li>
<li><a href="#orgheadline4">2.3. <span class="timestamp-wrapper"><span class="timestamp">[2015-11-17 Tue]</span></span></a></li>
<li><a href="#orgheadline5">2.4. <span class="timestamp-wrapper"><span class="timestamp">[2015-11-18 Wed]</span></span></a></li>
<li><a href="#orgheadline6">2.5. <span class="timestamp-wrapper"><span class="timestamp">[2015-11-20 Fri]</span></span></a></li>
<li><a href="#orgheadline7">2.6. <span class="timestamp-wrapper"><span class="timestamp">[2015-11-28 Sat]</span></span></a></li>
<li><a href="#orgheadline8">2.7. <span class="timestamp-wrapper"><span class="timestamp">[2015-12-03 Thu]</span></span></a></li>
</ul>
</li>
<li><a href="#orgheadline36">3. Scripts</a>
<ul>
<li><a href="#orgheadline29">3.1. Première version</a>
<ul>
<li><a href="#orgheadline10">3.1.1. Extraction des données brutes</a></li>
<li><a href="#orgheadline13">3.1.2. Des spectrogrammes aux données fastq non-trimmées</a>
<ul>
<li><a href="#orgheadline11">3.1.2.1. prérequis</a></li>
<li><a href="#orgheadline12">3.1.2.2. via seqret</a></li>
</ul>
</li>
<li><a href="#orgheadline14">3.1.3. Des données non-trimmées aux données trimmées et filtrées</a></li>
<li><a href="#orgheadline15">3.1.4. Analyse de la qualité des séquences</a></li>
<li><a href="#orgheadline16">3.1.5. Détermination des SNP</a></li>
<li><a href="#orgheadline28">3.1.6. Analyse des variants</a>
<ul>
<li><a href="#orgheadline17">3.1.6.1. Lecture des données et nettoyage</a></li>
<li><a href="#orgheadline18">3.1.6.2. Détermine la qualité des SNP</a></li>
<li><a href="#orgheadline19">3.1.6.3. Graphiques globaux</a></li>
<li><a href="#orgheadline20">3.1.6.4. Observations globales</a></li>
<li><a href="#orgheadline21">3.1.6.5. Détermination des SNP calibrés <code>[2/2]</code></a></li>
<li><a href="#orgheadline22">3.1.6.6. Positions terminales de switch <code>[1/3]</code></a></li>
<li><a href="#orgheadline23">3.1.6.7. SNP inattendus <code>[1/1]</code></a></li>
<li><a href="#orgheadline24">3.1.6.8. Position des SNP par rapport à la conversion tract <code>[1/1]</code></a></li>
<li><a href="#orgheadline27">3.1.6.9. Polymorphisme <code>[1/2]</code></a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#orgheadline35">3.2. Deuxième version</a>
<ul>
<li><a href="#orgheadline30">3.2.1. Base calling</a></li>
<li><a href="#orgheadline31">3.2.2. Alignement</a></li>
<li><a href="#orgheadline32">3.2.3. Pooling</a></li>
<li><a href="#orgheadline33">3.2.4. Sens des mutations</a></li>
<li><a href="#orgheadline34">3.2.5. Analyses</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#orgheadline37">4. Script Principal</a></li>
<li><a href="#orgheadline38">5. configuration&#xa0;&#xa0;&#xa0;<span class="tag"><span class="CONFIG">CONFIG</span></span></a></li>
</ul>
</div>
</div>
<div class="org-src-container">

<pre class="src src-emacs-lisp"><span style="color: #3a81c3;">(</span>org-babel-tangle-file <span style="color: #2d9574;">"README.org"</span><span style="color: #3a81c3;">)</span>
</pre>
</div>

<p>
Le dossier contient les données du séquençage des deux plaques envoyées à la
suite des manips de Florence. 
</p>

<div id="outline-container-orgheadline1" class="outline-2">
<h2 id="orgheadline1"><span class="section-number-2">1</span> Structure</h2>
<div class="outline-text-2" id="text-1">
<div class="org-src-container">

<pre class="src src-sh">tree ./ -L <span style="color: #4e3163;">2</span>
</pre>
</div>


<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">


<colgroup>
<col  class="org-left" />

<col  class="org-left" />

<col  class="org-left" />
</colgroup>
<thead>
<tr>
<th scope="col" class="org-left">dossier</th>
<th scope="col" class="org-left">fichier ou dossier</th>
<th scope="col" class="org-left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td class="org-left">analysis</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; plots</td>
<td class="org-left">dossier contenant les différents plots générés par plots-bcftools.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; snp<sub>distribution.pdf</sub></td>
<td class="org-left">la distribution des SNP sans tenir compte de la provenance des mutants</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; snp<sub>resume.pdf</sub></td>
<td class="org-left">les trois plots sur la même feuille</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; substitution<sub>distribution.pdf</sub></td>
<td class="org-left">la distribution des substitutions.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; trimmed<sub>fastqc.html</sub></td>
<td class="org-left">le contrôle de la qualité des séquences trimmées via fastqc</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; untrimmed<sub>fastqc.html</sub></td>
<td class="org-left">le contrôle de la qualité des séquences non-trimmées via fastqc</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; vincent<sub>plot.pdf</sub></td>
<td class="org-left">le graphe de vincent le jour de la réception des données.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; Analysis<sub>Summary</sub>-Sanger<sub>Pipeline.pdf</sub></td>
<td class="org-left">description du fichier d'analyse GATK</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-left">data</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; 1369607.INDEL.csv</td>
<td class="org-left">le fichier envoyé par GATK</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; 1369607.SNP.csv</td>
<td class="org-left">le fichier envoyé par GATK</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; 1369628.INDEL.csv</td>
<td class="org-left">le fichier envoyé par GATK</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; 1369628.SNP.csv</td>
<td class="org-left">le fichier envoyé par GATK</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; all.fasta</td>
<td class="org-left">toutes les séquences</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; fasta</td>
<td class="org-left">dossier contenant les séquences une à une</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; fastq</td>
<td class="org-left">idem en fastq</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; id<sub>table.dat</sub></td>
<td class="org-left">une table contenant les nom de séquence, le mutant, et la qualité du mutant</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; raw<sub>seq</sub><sub>nvbr</sub></td>
<td class="org-left">le dossier contenant les données brutes. NEPASTOUCHER</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; reference.fasta</td>
<td class="org-left">la séquence de référence</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; refseq.fasta</td>
<td class="org-left">idem</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; refseq<sub>reverse.fasta</sub></td>
<td class="org-left">la séquence de référence reversed.</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; seq</td>
<td class="org-left">le dossier contentant les séquneces</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; spectrograms</td>
<td class="org-left">le dossier contentant les .ab1 files</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; tmp</td>
<td class="org-left">un dossier de travail</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; trimmed.fasta</td>
<td class="org-left">les séquences trimmées</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; trimmed.fastq</td>
<td class="org-left">idem</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; untrimmed.fastq</td>
<td class="org-left">les séquences non trimmées</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; variantCalling</td>
<td class="org-left">le dossier de travail pour l'analyse des variants</td>
</tr>
</tbody>
<tbody>
<tr>
<td class="org-left">scripts</td>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#xa0;</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; ab1<sub>parser.py</sub></td>
<td class="org-left">convertit l'ensemble des fichiers .ab1 en fastq</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; ab1<sub>to</sub><sub>fastq</sub></td>
<td class="org-left">idem, utilitaire pipeable</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; exploratory<sub>analysis.R</sub></td>
<td class="org-left">premières analyses dans R</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; extract<sub>raw</sub><sub>data.sh</sub></td>
<td class="org-left">met en place la structure de données</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; make<sub>id</sub><sub>table.py</sub></td>
<td class="org-left">crée le fichier ../data/id<sub>table.dat</sub></td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; quality<sub>check</sub></td>
<td class="org-left">analyse la qualité via fastqc</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; trim<sub>low</sub><sub>quality.sh</sub></td>
<td class="org-left">trimme les séquences via bbduk</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; variantCallerBwa.sh</td>
<td class="org-left">un premier essai d'alignement et de snp calling via samtools et bcftools</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; variantCallerSsaha2.sh</td>
<td class="org-left">l'alignement avec ssaha2SNP</td>
</tr>

<tr>
<td class="org-left">&#xa0;</td>
<td class="org-left">&#x2013; variant<sub>analysis.R</sub></td>
<td class="org-left">l'analyse des variants et les graphes qui vont avec</td>
</tr>
</tbody>
</table>
</div>
</div>

<div id="outline-container-orgheadline9" class="outline-2">
<h2 id="orgheadline9"><span class="section-number-2">2</span> Commentaires</h2>
<div class="outline-text-2" id="text-2">
</div><div id="outline-container-orgheadline2" class="outline-3">
<h3 id="orgheadline2"><span class="section-number-3">2.1</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-11-09 Mon]</span></span></h3>
<div class="outline-text-3" id="text-2-1">
<p>
Pas de données dans le fichier <a href="./data/fasta/pS6-1073.fas">./data/fasta/pS6-1073.fas</a>, mais pourtant le
fichier <a href="./data/spectrograms/pS6-1073.ab1">./data/spectrograms/pS6-1073.ab1</a> en contient. On repart des .ab1 avec un
script <a href="./scripts/ab1_parser.py">./scripts/ab1_parser.py</a>, qui convertit les .ab1 en fasta et fastq. 
</p>

<p>
Le fichier <a href="./data/fasta/pS6-1073.fst">./data/fasta/pS6-1073.fst</a> est toujours aussi mauvais. Erreurs de
séquençage ? À exclure des analyses. 
</p>

<p>
Le fichier <a href="./data/fasta/pS9-1073.fst">./data/fasta/pS9-1073.fst</a> a un indel en position 343-342 et en
position 397. Au vu du spectrogramme <a href="./data/spectrograms/pS9-1073.ab1">./data/spectrograms/pS9-1073.ab1</a>, c'est une
erreur de séquençage. Globalement, qualité du séquençage pas très bonne. À
exclure des analyses.
</p>
</div>
</div>
<div id="outline-container-orgheadline3" class="outline-3">
<h3 id="orgheadline3"><span class="section-number-3">2.2</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-11-16 Mon]</span></span></h3>
<div class="outline-text-3" id="text-2-2">
<p>
La semaine dernière, le script <a href="./scripts/ab1_parser.py">./scripts/ab1_parser.py</a> convertissait tous les
spectrograms de <code>ab1</code> vers <code>fastq</code>. Combiné en 1 fichier, <code>all.fastq</code>, on a
utilisé <i>fastqc</i> pour avoir une idée de la qualité. Le résultat dans
<a href="./analysis/all_fastqc.html">./analysis/all_fastqc.html</a>. Il a été convenu arbitrairement avec Vincent que les bases d'une
qualité &lt; 28 seraient exclues de l'analyse. Aujourd'hui, le script
<a href="./scripts/ab1_parser.py">./scripts/ab1_parser.py</a> va être modifié en conséquence. Voir les anciennes
versions via <i>git</i> éventuellement.
</p>

<p>
J'écris également aujourd'hui le script <a href="./scripts/trim_low_quality.py">./scripts/trim_low_quality.py</a>, qui
enlève les 30 premières et dernières séquences, et qui empêche les séquences de
trop mauvaise qualité d'être utilisées dans l'analyse. 
</p>

<p>
Finalement, inutile de réinventer la roue. <code>Fastx_toolkit</code> devrait normalement
faire ça très bien, mais ça ne fonctionne pas, pour des raisons que je ne
m'explique pas. Cependant, je suis tombé sur l'utilitaire <code>BBmap</code>, qui contient
entre autre, <code>bbduk</code>. <a href="file:///Users/samuelbarreto/.bin/bbmap/bbduk.sh">Voir le fichier source ici</a>, la page de téléchargement <a href="http://sourceforge.net/projects/bbmap/?source=typ_redirect">là</a>,
et pour des commentaires sur l'utilisation <a href="http://seqanswers.com/forums/showthread.php?t=58221">Voir là</a>, et <a href="http://seqanswers.com/forums/showthread.php?t=42776">là</a>.
</p>
</div>
</div>

<div id="outline-container-orgheadline4" class="outline-3">
<h3 id="orgheadline4"><span class="section-number-3">2.3</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-11-17 Tue]</span></span></h3>
<div class="outline-text-3" id="text-2-3">
<p>
Je supprime donc le script <code>low_quality_trim.py</code>. J'utilise le script
<code>low_quality_trim.sh</code>. 
</p>

<p>
En résultat, comparer <a href="./analysis/trimmed_fastqc.html">./analysis/trimmed_fastqc.html</a> et
<a href="./analysis/untrimmed_fastqc.html">./analysis/untrimmed_fastqc.html</a>. On n'a plus que 179 séquences au lieu de 192,
mais ça vaut le coup, la qualité est largement supérieure. 
</p>

<p>
Je veux maintenant déterminer les SNPs. Il faut donc que j'aligne les séquences
obtenues avec la séquence de référence <a href="./data/refseq.fasta">./data/refseq.fasta</a>. GATC utilise le
software <code>SSAHA2</code> (voir <a href="http://www.sanger.ac.uk/science/tools/ssaha2-0">là</a>) mais à priori il n'est plus utilisable. Le site
recommande d'utiliser <code>SMALT</code>, (voir la page de téléchargement <a href="http://sourceforge.net/projects/smalt/?source=typ_redirect">là</a>, le manuel <a href="ftp://ftp.sanger.ac.uk/pub/resources/software/smalt/smalt-manual-0.7.4.pdf">là</a>
et la page du software <a href="http://www.sanger.ac.uk/science/tools/smalt-0">là</a>. 
</p>

<p>
En fait, je l'ai juste installé comme ça :
</p>

<div class="org-src-container">

<pre class="src src-sh">brew update
brew tap homebrew/homebrew-science
brew install smalt
</pre>
</div>

<p>
Finalement, c'est encore un autre workflow que je veux adopter. On repart sur
<code>ssaha2</code> et <code>ssaha2SNP</code>, la page de téléchargement étant
<a href="ftp://ftp.sanger.ac.uk/pub/resources/software/ssaha2/">ftp://ftp.sanger.ac.uk/pub/resources/software/ssaha2/</a>.  
</p>
</div>
</div>
<div id="outline-container-orgheadline5" class="outline-3">
<h3 id="orgheadline5"><span class="section-number-3">2.4</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-11-18 Wed]</span></span></h3>
<div class="outline-text-3" id="text-2-4">
<p>
Il faut clarifier les étapes permettant d'aligner et de déterminer les SNP. Ce
qui est fait dans le script <a href="./scripts/variantCallerSsaha2.sh">./scripts/variantCallerSsaha2.sh</a>. 
</p>
</div>
</div>
<div id="outline-container-orgheadline6" class="outline-3">
<h3 id="orgheadline6"><span class="section-number-3">2.5</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-11-20 Fri]</span></span></h3>
<div class="outline-text-3" id="text-2-5">
<p>
Le rapport était basé sur une version de ma fonction <code>mutant_caller</code> dans le
script R <a href="./scripts/variant_analysis.R">./scripts/variant_analysis.R</a> qui était fausse. Très fausse. Beaucoup de
boulot à corriger. 
</p>
</div>
</div>

<div id="outline-container-orgheadline7" class="outline-3">
<h3 id="orgheadline7"><span class="section-number-3">2.6</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-11-28 Sat]</span></span></h3>
<div class="outline-text-3" id="text-2-6">
<p>
La photo suivante montre le spectrogramme d'une position appelée N dans les
alignements. Pour moi c'est la marque de l'hétérogénéité dans la colonie. Les
uns réparent en G, les autres en A ou T.
</p>


<div class="figure">
<p><img src="./analysis/litigieux_snp.png" alt="litigieux_snp.png" /> 
</p>
</div>
</div>
</div>

<div id="outline-container-orgheadline8" class="outline-3">
<h3 id="orgheadline8"><span class="section-number-3">2.7</span> <span class="timestamp-wrapper"><span class="timestamp">[2015-12-03 Thu]</span></span></h3>
<div class="outline-text-3" id="text-2-7">
<p>
J'ai cherché à extraire les données trace par trace, mais à priori il n'y a pas
d'OSS qui permette de faire ça. On va donc utiliser la qualité comme proxy du
polymorphisme. Ça implique de baisser le seuil d'exigence quant aux bases qu'on
décide de renommer <code>N</code>. Donc modification du script 
</p>
</div>
</div>
</div>
<div id="outline-container-orgheadline36" class="outline-2">
<h2 id="orgheadline36"><span class="section-number-2">3</span> Scripts</h2>
<div class="outline-text-2" id="text-3">
</div><div id="outline-container-orgheadline29" class="outline-3">
<h3 id="orgheadline29"><span class="section-number-3">3.1</span> Première version</h3>
<div class="outline-text-3" id="text-3-1">
<p>
Contient les différents scripts nécessaires pour aboutir aux contenus
d'<code>analysis</code>. 
</p>
</div>

<div id="outline-container-orgheadline10" class="outline-4">
<h4 id="orgheadline10"><span class="section-number-4">3.1.1</span> Extraction des données brutes</h4>
<div class="outline-text-4" id="text-3-1-1">
<p>
Pour extraire les données du fichier brut .zip à la structure de données.
</p>

<div class="org-src-container">

<pre class="src src-sh"><span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">!/bin/</span><span style="color: #3a81c3; font-weight: bold;">bash</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>

<span style="color: #3a81c3;">cd</span> ~/stage/seq_novembre/
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">Le script qui extrait les donn&#233;es depuis les fichiers zip bruts et qui met en</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">place la structure de fichier.</span>

<span style="color: #3a81c3;">cd</span> ./data
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">extraction des donn&#233;es brutes</span>
unzip raw_seq_nvbr/1369607.zip
unzip raw_seq_nvbr/1369628.zip

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">cr&#233;e les dossiers</span>
mkdir fasta seq spectrograms csv
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">d&#233;place tout les fichiers dans des dossiers adapt&#233;s </span>
find . -name <span style="color: #2d9574;">"*.fas"</span> -exec mv -i <span style="color: #3a81c3;">{}</span> -t ./fasta/ <span style="color: #2d9574;">\;</span>
find . -name <span style="color: #2d9574;">"*.ab1"</span> -exec mv -i <span style="color: #3a81c3;">{}</span> -t ./spectrograms/ <span style="color: #2d9574;">\;</span>
find . -name <span style="color: #2d9574;">"*.seq"</span> -exec mv -i <span style="color: #3a81c3;">{}</span> -t ./seq/ <span style="color: #2d9574;">\;</span>
find . -name <span style="color: #2d9574;">"*.csv"</span> -exec mv -i <span style="color: #3a81c3;">{}</span> -t ./csv/ <span style="color: #2d9574;">\;</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">d&#233;place le contenu du dossier inutile dans le pr&#233;sent dossier</span>
mv <span style="color: #4e3163;">1369628/*</span> ./
rm -r <span style="color: #4e3163;">1369628</span> <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">supprime le dossier</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">d&#233;place le fichier pdf dans le dossier adapt&#233;</span>
mv *.pdf ../analysis
</pre>
</div>

<p>
Les spectrogrammes contiennent l'info de la sequence<sub>id</sub> et du nom. 
On construit une table avec la qualité du mutant en troisième colonne. 
</p>

<div class="org-src-container">

<pre class="src src-python"><span style="color: #3a81c3; font-weight: bold;">from</span> Bio <span style="color: #3a81c3; font-weight: bold;">import</span> SeqIO
<span style="color: #3a81c3; font-weight: bold;">import</span> glob

<span style="color: #3a81c3; font-weight: bold;">def</span> <span style="color: #6c3163; font-weight: bold;">strong_or_weak</span><span style="color: #3a81c3;">(</span>record<span style="color: #3a81c3;">)</span>:
    <span style="color: #2d9574;">"""</span>
<span style="color: #2d9574;">    Determine si le mutant est strong ou weak</span>
<span style="color: #2d9574;">    """</span>
    <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #2d9574;">'S'</span> <span style="color: #3a81c3; font-weight: bold;">in</span> record:
        <span style="color: #3a81c3; font-weight: bold;">return</span> <span style="color: #2d9574;">'strong'</span>
    <span style="color: #3a81c3; font-weight: bold;">else</span>:
        <span style="color: #3a81c3; font-weight: bold;">return</span> <span style="color: #2d9574;">'weak'</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">en-tete de colonne</span>
<span style="color: #3a81c3; font-weight: bold;">print</span> <span style="color: #2d9574;">"id name mutant"</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">pour chaque fichier ab1</span>
<span style="color: #3a81c3; font-weight: bold;">for</span> <span style="color: #3a81c3;">file</span> <span style="color: #3a81c3; font-weight: bold;">in</span> glob.glob<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"../data/spectrograms/*.ab1"</span><span style="color: #3a81c3;">)</span>:
    <span style="color: #3a81c3; font-weight: bold;">with</span> <span style="color: #3a81c3;">open</span><span style="color: #3a81c3;">(</span><span style="color: #3a81c3;">file</span>, <span style="color: #2d9574;">"rb"</span><span style="color: #3a81c3;">)</span> <span style="color: #3a81c3; font-weight: bold;">as</span> spectro:
        <span style="color: #3a81c3; font-weight: bold;">for</span> record <span style="color: #3a81c3; font-weight: bold;">in</span> SeqIO.parse<span style="color: #3a81c3;">(</span>spectro, <span style="color: #2d9574;">"abi"</span><span style="color: #3a81c3;">)</span>:
            <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">associer l'id avec le nom et le type de mutant</span>
            <span style="color: #3a81c3; font-weight: bold;">print</span> record.<span style="color: #3a81c3;">id</span> + <span style="color: #2d9574;">" "</span> + record.name + \
                <span style="color: #2d9574;">" "</span> + strong_or_weak<span style="color: #3a81c3;">(</span>record.name<span style="color: #3a81c3;">)</span>
</pre>
</div>

<p>
On crée la table en question via :
</p>
<div class="org-src-container">

<pre class="src src-sh"><span style="color: #3a81c3;">cd</span> ~/stage/seq_novembre/scripts
python make_id_table.py &gt; ../data/id_table.dat
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline13" class="outline-4">
<h4 id="orgheadline13"><span class="section-number-4">3.1.2</span> Des spectrogrammes aux données fastq non-trimmées</h4>
<div class="outline-text-4" id="text-3-1-2">
</div><div id="outline-container-orgheadline11" class="outline-5">
<h5 id="orgheadline11"><span class="section-number-5">3.1.2.1</span> prérequis</h5>
<div class="outline-text-5" id="text-3-1-2-1">
<div class="org-src-container">

<pre class="src src-sh">brew install emboss
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline12" class="outline-5">
<h5 id="orgheadline12"><span class="section-number-5">3.1.2.2</span> via seqret</h5>
<div class="outline-text-5" id="text-3-1-2-2">
<div class="org-src-container">

<pre class="src src-sh"><span style="color: #3a81c3;">cd</span> ~/stage/seq_novembre/scripts

touch untrimmed.fastq
<span style="color: #3a81c3; font-weight: bold;">for</span> file<span style="color: #3a81c3; font-weight: bold;"> in</span> ../data/spectrograms/*.ab1
<span style="color: #3a81c3; font-weight: bold;">do</span>
    seqret <span style="color: #2d9574;">\</span>
        -sformat abi <span style="color: #2d9574;">\</span>
        -osformat fastq <span style="color: #2d9574;">\</span>
        -auto <span style="color: #2d9574;">\</span>
        -stdout <span style="color: #2d9574;">\</span>
        -sequence $<span style="color: #715ab1;">file</span> <span style="color: #2d9574;">\</span>
        &gt;&gt; ../data/untrimmed.fastq
<span style="color: #3a81c3; font-weight: bold;">done</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">convertit le fastq en fasta</span>
seqret -sformat fastq -osformat fasta -auto -stdout <span style="color: #2d9574;">\</span>
       -sequence ../data/untrimmed.fastq &gt; ../data/untrimmed.fasta
</pre>
</div>
</div>
</div>
</div>

<div id="outline-container-orgheadline14" class="outline-4">
<h4 id="orgheadline14"><span class="section-number-4">3.1.3</span> Des données non-trimmées aux données trimmées et filtrées</h4>
<div class="outline-text-4" id="text-3-1-3">
<p>
Un script qui convertit le fichier <a href="./data/untrimmed.fastq">./data/untrimmed.fastq</a> en fichier
<a href="./data/trimmed.fastq">./data/trimmed.fastq</a>. <i>bbduk</i> trimme les bases de faible qualité. seqtk
convertit les bases restantes de faible qualité en N, seqret convertit le fastq
généré en fasta. 
</p>

<div class="org-src-container">

<pre class="src src-sh"><span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">!/usr/local/bin/</span><span style="color: #3a81c3; font-weight: bold;">bash</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' -qtrim=rl : quality trim right and left </span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' -trimq=28 : trim if quality &lt; 28 (sanger encoding, illumina 1.9)</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' -minlen=620 : keep only seq with length &gt; 620, after trimming.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' -Xmx1g : tells bbduk / java to use 1G of RAM</span>

<span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #3a81c3;">[</span><span style="color: #6c3163;">[</span> -f ../data/untrimmed.fastq &amp;&amp; <span style="color: #4e3163;">!</span> -f ../data/trim.fastq <span style="color: #6c3163;">]</span><span style="color: #3a81c3;">]</span>
<span style="color: #3a81c3; font-weight: bold;">then</span> <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">si les fichiers n'existent pas.</span>
    ~/.bin/bbmap/bbduk.sh -Xmx1g <span style="color: #2d9574;">\</span>
                          -in=../data/untrimmed.fastq <span style="color: #2d9574;">\</span>
                          -out=../data/trim.fastq <span style="color: #2d9574;">\</span>
                          -qtrim=rl <span style="color: #2d9574;">\</span>
                          -trimq=<span style="color: #4e3163;">28</span> <span style="color: #2d9574;">\</span>
                          -minlen=<span style="color: #4e3163;">620</span>

    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">convertit les bases d'une qualit&#233; inf&#233;rieure &#224; 20 en N.</span>
    seqtk seq -q20 -nN ../data/trim.fastq &gt; ../data/trimmed.fastq

    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">convertit le fastq en fasta</span>
    seqret -sformat fastq -osformat fasta -auto -stdout <span style="color: #2d9574;">\</span>
           -sequence ../data/trimmed.fastq &gt; ../data/trimmed.fasta

    rm ../data/trim.fastq
<span style="color: #3a81c3; font-weight: bold;">else</span>
    <span style="color: #3a81c3;">printf</span> <span style="color: #2d9574;">"Le fichier untrimmed.fastq n'existe pas, ou le fichier trimmed.fastq existe d&#233;j&#224;."</span>
<span style="color: #3a81c3; font-weight: bold;">fi</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline15" class="outline-4">
<h4 id="orgheadline15"><span class="section-number-4">3.1.4</span> Analyse de la qualité des séquences</h4>
<div class="outline-text-4" id="text-3-1-4">
<p>
Le script utilisé pour analyser les données de qualité via <i>fastqc</i>. 
</p>

<div class="org-src-container">

<pre class="src src-sh"><span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">!/usr/local/bin/</span><span style="color: #3a81c3; font-weight: bold;">bash</span>
<span style="color: #3a81c3;">cd</span> ~/stage/seq_novembre/scripts

<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">quand dans le dossier ./scripts</span>
<span style="color: #3a81c3;">cd</span> ../data/

<span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #3a81c3;">[</span> -f untrimmed.fastq <span style="color: #3a81c3;">]</span> &amp;&amp; <span style="color: #3a81c3;">[</span> -f trimmed.fastq <span style="color: #3a81c3;">]</span> ; <span style="color: #3a81c3; font-weight: bold;">then</span>
    mkdir tmp
    <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">analyse les donn&#233;es et output dans tmp</span>
    fastqc untrimmed.fastq -o ./tmp
    fastqc trimmed.fastq   -o ./tmp
    <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">unzip resulting files</span>
    unzip -qq tmp/untrimmed_fastqc.zip -d tmp
    unzip -qq tmp/trimmed_fastqc.zip -d tmp
    <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">extract main results</span>
    mv tmp/untrimmed_fastqc/Images/per_base_quality.png <span style="color: #2d9574;">\</span>
       ../analysis/per_base_quality_fastqc_untrimmed.png
    mv tmp/trimmed_fastqc/Images/per_base_quality.png <span style="color: #2d9574;">\</span>
       ../analysis/per_base_quality_fastqc_trimmed.png
    <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">copy html into analysis</span>
    mv tmp/*.html ../analysis/
    <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">delete tmp files</span>
    rm -r tmp <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">remove temporary files</span>

<span style="color: #3a81c3; font-weight: bold;">else</span>
    <span style="color: #3a81c3;">printf</span> <span style="color: #2d9574;">"Les fichiers untrimmed.fastq et trimmed.fastq n'existent pas."</span>
<span style="color: #3a81c3; font-weight: bold;">fi</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline16" class="outline-4">
<h4 id="orgheadline16"><span class="section-number-4">3.1.5</span> Détermination des SNP</h4>
<div class="outline-text-4" id="text-3-1-5">
<div class="org-src-container">

<pre class="src src-sh"><span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">!/bin/</span><span style="color: #3a81c3; font-weight: bold;">bash</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">variant calling using ssaha2 and ssaha2SNP</span>

<span style="color: #3a81c3;">cd</span> ~/stage/seq_novembre/data
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">prend le reverse complement de la s&#233;quence de r&#233;f&#233;rence</span>
fastx_reverse_complement -i reference.fasta -o reference_reverse.fasta

mkdir variantCalling
<span style="color: #3a81c3;">cd</span> variantCalling

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">place les s&#233;quences n&#233;cessaires pour l'analyse dans le dossier. </span>
ln -s ../trimmed.fastq .
ln -s ../reference_reverse.fasta ./reference_reverse.fasta

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">alignement &#224; la s&#233;quence de r&#233;f&#233;rence</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' -output psl :             format de sortie psl</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' reference_reverse.fasta : s&#233;quence de r&#233;f&#233;rence</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' trimmed.fastq :           s&#233;quence &#224; aligner</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">' output.psl :              fichier de sortie</span>
~/.bin/ssahaSNP/ssaha2 -output psl reference_reverse.fasta trimmed.fastq &gt; output.psl

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">polymorphism detection tool</span>
~/.bin/ssahaSNP/ssahaSNP reference_reverse.fasta trimmed.fastq &gt; SNP.txt

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">computer readable format conversion</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">egrep trouve les lignes o&#249; sont indiqu&#233;es les donn&#233;es concernant les SNP</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">awk extrait les champs en question dans un fichier SNP.dat</span>
egrep ssaha:SNP SNP.txt | <span style="color: #2d9574;">\</span>
    awk <span style="color: #2d9574;">'{print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15}'</span> &gt; SNP.dat

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">column annotation based on</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">ftp://ftp.sanger.ac.uk/pub/resources/software/ssahasnp/readme,</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">part (6) some further information</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">la premi&#232;re ligne du fichier .dat, afin d'&#234;tre lu dans R</span>
<span style="color: #3a81c3;">echo</span> <span style="color: #2d9574;">" match subject_name index_of_subject read_name s_base q_base s_qual q_qual offset_on_subject offset_on_read length_of_snp start_match_of_read end_match_of_read match_direction length_of_subject "</span> &gt; head.dat
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">into final document</span>
cat head.dat SNP.dat &gt; snp_calling.dat
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline28" class="outline-4">
<h4 id="orgheadline28"><span class="section-number-4">3.1.6</span> Analyse des variants</h4>
<div class="outline-text-4" id="text-3-1-6">
</div><div id="outline-container-orgheadline17" class="outline-5">
<h5 id="orgheadline17"><span class="section-number-5">3.1.6.1</span> Lecture des données et nettoyage</h5>
<div class="outline-text-5" id="text-3-1-6-1">
<p>
Le fichier <a href="file:///Users/samuelbarreto/stage/seq_novembre/data/variantCalling/snp_calling.dat">file:///Users/samuelbarreto/stage/seq_novembre/data/variantCalling/snp_calling.dat</a> contient les
données d'intérêt. Il ne contient ni le nom du transformant, ni son type (Weak
ou Strong). Les données sont récupérées dans le fichier <a href="./data/id_table.dat">./data/id_table.dat</a>. Les
deux tables sont associés via un <i>inner<sub>join</sub></i>. 
</p>

<p>
On s'est rendu compte que le plasmide <code>pS60</code> était un contaminant, toutes les
mutations qu'il engendre sont de type <i>weak</i>. Il est donc rebasculé dans le
tableau des Weak. 
</p>

<div class="org-src-container">

<pre class="src src-R">setwd<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"~/stage/seq_novembre/data/variantCalling"</span><span style="color: #3a81c3;">)</span>

<span style="color: #4e3163;">library</span><span style="color: #3a81c3;">(</span>dplyr<span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">read the data</span>
snp <span style="color: #4e3163;">&lt;-</span> tbl_df<span style="color: #3a81c3;">(</span>read.table<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"snp_calling.dat"</span>, head = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">enl&#232;ve les colonnes inutiles</span>
snp %&gt;%
    select<span style="color: #3a81c3;">(</span>
        -match, -subject_name, -index_of_subject, -length_of_subject,
        -match_direction, -contains<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"_of_read"</span><span style="color: #6c3163;">)</span>, -contains<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"on_read"</span><span style="color: #6c3163;">)</span>,
        -contains<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"_of_snp"</span><span style="color: #6c3163;">)</span>, -s_qual
    <span style="color: #3a81c3;">)</span> <span style="color: #4e3163;">-&gt;</span> snp

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">lit les m&#233;tadonn&#233;es de s&#233;quence</span>
id_table <span style="color: #4e3163;">&lt;-</span> tbl_df<span style="color: #3a81c3;">(</span>read.table<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"../id_table.dat"</span>, head = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">fait correspondre le read_name avec le nom du clone et le type de mutant W ou S</span>
snp <span style="color: #4e3163;">&lt;-</span> inner_join<span style="color: #3a81c3;">(</span>x = snp, y = id_table, by = c<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"read_name"</span> = <span style="color: #2d9574;">"id"</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">suppress tmp var</span>
rm<span style="color: #3a81c3;">(</span>id_table<span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">bascule les contaminants mysterieux dans la bonne cat&#233;gorie</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">TESTE ET APPROUVE</span>
snp$mutant<span style="color: #3a81c3;">[</span>snp$name == <span style="color: #2d9574;">"pS60-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
snp$mutant<span style="color: #3a81c3;">[</span>snp$name == <span style="color: #2d9574;">"pS83-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
snp$mutant<span style="color: #3a81c3;">[</span>snp$name == <span style="color: #2d9574;">"pS92-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
snp$mutant<span style="color: #3a81c3;">[</span>snp$name == <span style="color: #2d9574;">"pS91-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
snp$mutant<span style="color: #3a81c3;">[</span>snp$name == <span style="color: #2d9574;">"pW6-1073"</span> <span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"strong"</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline18" class="outline-5">
<h5 id="orgheadline18"><span class="section-number-5">3.1.6.2</span> Détermine la qualité des SNP</h5>
<div class="outline-text-5" id="text-3-1-6-2">
<p>
Pour déterminer si le SNP est de type <i>weak</i> ou <i>strong</i>, j'utilise la fonction
<code>mutant_caller</code>. Si la référence est A ou T, soit le transformant est C ou G, et
la substitution est <i>WS</i> ; soit le transformant est A ou T, et la mutation est
<i>WW</i>. Si la référence est C ou G, soit le transformant est A ou T, et la
substitution est <i>SW</i> ; soit le transformant est G ou C, la substitution est
<i>SS</i>.
</p>

<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> une fonction pour d&#233;terminer si la substitution est strong ou weak. On peut</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> avoir des substitutions weak chez les strongs</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">subject</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la base sur la s&#233;quence de r&#233;f&#233;rence</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">query</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la base sur le read.</span>
<span style="color: #6c3163; font-weight: bold;">mutant_caller</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>subject, query<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
    <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #6c3163;">(</span>subject == <span style="color: #2d9574;">'A'</span> || subject == <span style="color: #2d9574;">'T'</span><span style="color: #6c3163;">)</span> <span style="color: #6c3163;">{</span>
        <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #2d9574;">(</span>query == <span style="color: #2d9574;">'C'</span> || query == <span style="color: #2d9574;">'G'</span> <span style="color: #2d9574;">)</span> <span style="color: #2d9574;">{</span>
            <span style="color: #2d9574;">'WS'</span>
        <span style="color: #2d9574;">}</span> <span style="color: #3a81c3; font-weight: bold;">else</span> <span style="color: #2d9574;">{</span>
            <span style="color: #2d9574;">'WW'</span>
        <span style="color: #2d9574;">}</span>
    <span style="color: #6c3163;">}</span> <span style="color: #3a81c3; font-weight: bold;">else</span> <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #6c3163;">(</span>subject == <span style="color: #2d9574;">'C'</span> || subject == <span style="color: #2d9574;">'G'</span><span style="color: #6c3163;">)</span> <span style="color: #6c3163;">{</span>
        <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #2d9574;">(</span>query == <span style="color: #2d9574;">'A'</span> || query == <span style="color: #2d9574;">'T'</span><span style="color: #2d9574;">)</span> <span style="color: #2d9574;">{</span>
            <span style="color: #2d9574;">'SW'</span>
        <span style="color: #2d9574;">}</span> <span style="color: #3a81c3; font-weight: bold;">else</span> <span style="color: #2d9574;">{</span>
            <span style="color: #2d9574;">'SS'</span>
        <span style="color: #2d9574;">}</span>
    <span style="color: #6c3163;">}</span>
<span style="color: #3a81c3;">}</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">on applique la fonction rowwise, ie ligne par ligne, via `mutate`, puis on</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">d&#233;groupe.</span>
snp %&gt;%
    rowwise<span style="color: #3a81c3;">()</span> %&gt;%
    mutate<span style="color: #3a81c3;">(</span>mutation_type = mutant_caller<span style="color: #6c3163;">(</span>s_base, q_base<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
    ungroup<span style="color: #3a81c3;">()</span> <span style="color: #4e3163;">-&gt;</span>
    snp
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">conversion en facteur</span>
snp$mutation_type = factor<span style="color: #3a81c3;">(</span>snp$mutation_type<span style="color: #3a81c3;">)</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-orgheadline19" class="outline-5">
<h5 id="orgheadline19"><span class="section-number-5">3.1.6.3</span> Graphiques globaux</h5>
<div class="outline-text-5" id="text-3-1-6-3">
<p>
Des graphiques de distribution globale des SNP sont fait ici. 
</p>

<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">SHORTCUT PLOT FUNCTION</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>

<span style="color: #4e3163;">library</span><span style="color: #3a81c3;">(</span>ggplot2<span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Une fonction qui permet de court-circuiter ggplot : repr&#233;sente en ordonn&#233;e</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la distribution des positions de snp, en abscisse la position des SNPs, par</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> d&#233;fault la couleur rep&#233;resente le type de mutant, peut &#234;tre &#233;galement</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> attribu&#233;e &#224; mutation_type. Respecte les crit&#232;res visuels de tufte. N&#233;cessite</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> ggplot2 1.02 si je ne m'abuse, avec l'option panel.ontop en tout cas.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> plot_snp</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">snp</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> les donn&#233;es de snp</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">fill</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la couleur par laquelle on color les barres</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">legend_position</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la position de la l&#233;gende sur le graphe</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">legend_name</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> le titre de la l&#233;gende. rien par d&#233;fault</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> un graphique</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel Barreto</span>
<span style="color: #6c3163; font-weight: bold;">plot_snp</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>data, fill_by = <span style="color: #2d9574;">"mutant"</span>,
                                  legend_position = c<span style="color: #6c3163;">(</span>0.2, 0.8<span style="color: #6c3163;">)</span>,
                                  legend_name = <span style="color: #2d9574;">""</span><span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
    plot <span style="color: #4e3163;">&lt;-</span> ggplot<span style="color: #6c3163;">(</span>data = data, aes<span style="color: #2d9574;">(</span>offset_on_subject<span style="color: #2d9574;">)</span><span style="color: #6c3163;">)</span> +
        theme_minimal<span style="color: #6c3163;">(</span>base_family = <span style="color: #2d9574;">"Courier"</span><span style="color: #6c3163;">)</span> +
        scale_x_continuous<span style="color: #6c3163;">(</span>breaks = seq<span style="color: #2d9574;">(</span>1, 734, 30<span style="color: #2d9574;">)</span><span style="color: #6c3163;">)</span> +
        scale_fill_brewer<span style="color: #6c3163;">(</span>palette = <span style="color: #2d9574;">"Set2"</span>, name = legend_name<span style="color: #6c3163;">)</span> +
        xlab<span style="color: #6c3163;">(</span><span style="color: #2d9574;">""</span><span style="color: #6c3163;">)</span> +
        ylab<span style="color: #6c3163;">(</span><span style="color: #2d9574;">""</span><span style="color: #6c3163;">)</span> +
        theme<span style="color: #6c3163;">(</span>panel.ontop = <span style="color: #ba2f59; font-weight: bold;">TRUE</span>, legend.position = legend_position,
              axis.text = element_text<span style="color: #2d9574;">(</span>size = 8, colour = <span style="color: #2d9574;">"gray"</span><span style="color: #2d9574;">)</span>,
              panel.grid.major.x = element_blank<span style="color: #2d9574;">()</span>,
              panel.grid.minor.x = element_blank<span style="color: #2d9574;">()</span>,
              panel.grid.minor.y = element_blank<span style="color: #2d9574;">()</span>,
              panel.grid.major.y = element_line<span style="color: #2d9574;">(</span>colour = <span style="color: #2d9574;">"white"</span>, size = 1<span style="color: #2d9574;">)</span><span style="color: #6c3163;">)</span>

    <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #6c3163;">(</span>fill_by == <span style="color: #2d9574;">"mutation_type"</span><span style="color: #6c3163;">)</span>
    <span style="color: #6c3163;">{</span>
        plot + geom_histogram<span style="color: #2d9574;">(</span>aes<span style="color: #67b11d;">(</span>fill = mutation_type<span style="color: #67b11d;">)</span>, binwidth = 10,
                              position = <span style="color: #2d9574;">"dodge"</span><span style="color: #2d9574;">)</span>
    <span style="color: #6c3163;">}</span> <span style="color: #3a81c3; font-weight: bold;">else</span> <span style="color: #6c3163;">{</span>
        plot + geom_histogram<span style="color: #2d9574;">(</span>aes<span style="color: #67b11d;">(</span>fill = mutant<span style="color: #67b11d;">)</span>, binwidth = 10,
                              position = <span style="color: #2d9574;">"dodge"</span><span style="color: #2d9574;">)</span>
    <span style="color: #6c3163;">}</span>
<span style="color: #3a81c3;">}</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">PLOT DISTRIBUTIONS</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>

pdf<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/snp_distribution.pdf"</span>, height = 5.8, width = 8.3<span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">distribution des SNP</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">fac&#233;t&#233;e par type de mutant, couleur = type de mutation</span>
snp %&gt;%
    plot_snp<span style="color: #3a81c3;">(</span>legend_name = <span style="color: #2d9574;">"Exogene"</span>, legend_pos= c<span style="color: #6c3163;">(</span>.2, .8<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>

dev.off<span style="color: #3a81c3;">()</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">-------------------------------------------------------------------------------</span>
pdf<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/mutant_snp_distribution.pdf"</span>, height = 5.8, width = 8.3<span style="color: #3a81c3;">)</span>

snp %&gt;%
    plot_snp<span style="color: #3a81c3;">(</span>fill_by = <span style="color: #2d9574;">"mutation_type"</span>, legend_name = <span style="color: #2d9574;">"Type de Mutation"</span> <span style="color: #3a81c3;">)</span> +
    facet_grid<span style="color: #3a81c3;">(</span>mutant ~ .<span style="color: #3a81c3;">)</span>

dev.off<span style="color: #3a81c3;">()</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline20" class="outline-5">
<h5 id="orgheadline20"><span class="section-number-5">3.1.6.4</span> Observations globales</h5>
<div class="outline-text-5" id="text-3-1-6-4">
<p>
Différents tableaux de résumé généraux. 
</p>

<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">=============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">OBSERVATIONS G&#201;N&#201;RALES</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">=============================================================================</span>

sink<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/observations.tex"</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    group_by<span style="color: #3a81c3;">(</span>mutant, name<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>mean = mean<span style="color: #6c3163;">(</span>count<span style="color: #6c3163;">)</span>, med = median<span style="color: #6c3163;">(</span>count<span style="color: #6c3163;">)</span>, sd = sd<span style="color: #6c3163;">(</span>count<span style="color: #6c3163;">)</span> <span style="color: #3a81c3;">)</span> %&gt;%
    knitr::kable<span style="color: #3a81c3;">(</span> align = <span style="color: #2d9574;">'c'</span>, digits = 2, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span>, format = <span style="color: #2d9574;">"latex"</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>

sink<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/count_by_mutant.tex"</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    group_by<span style="color: #3a81c3;">(</span>mutant<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    knitr::kable<span style="color: #3a81c3;">(</span> align = <span style="color: #2d9574;">'c'</span>, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span>, format = <span style="color: #2d9574;">"latex"</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>

sink<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/count_by_muttype.tex"</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    group_by<span style="color: #3a81c3;">(</span>mutation_type<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    knitr::kable<span style="color: #3a81c3;">(</span>col.names = c<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"Type de mutation"</span>, <span style="color: #2d9574;">"nombre"</span><span style="color: #6c3163;">)</span>,
                 align = <span style="color: #2d9574;">'c'</span>,
                 booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span>, format = <span style="color: #2d9574;">"latex"</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>

sink<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/seq_by_mutant.tex"</span><span style="color: #3a81c3;">)</span>
distinct<span style="color: #3a81c3;">(</span>snp, name, mutant<span style="color: #3a81c3;">)</span> %&gt;%
    group_by<span style="color: #3a81c3;">(</span>mutant<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    knitr::kable<span style="color: #3a81c3;">(</span> align = <span style="color: #2d9574;">'c'</span>, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span>, format = <span style="color: #2d9574;">"latex"</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline21" class="outline-5">
<h5 id="orgheadline21"><span class="section-number-5">3.1.6.5</span> Détermination des SNP calibrés <code>[2/2]</code></h5>
<div class="outline-text-5" id="text-3-1-6-5">
<ul class="org-ul">
<li class="on"><code>[X]</code> créer la fonction <code>is_a_position</code> qui détermine si un SNP est bien à la
position attendu</li>
<li class="on"><code>[X]</code> faire la même chose par type de mutant</li>
</ul>
<p>
On veut filtrer les positions qui sont bien les SNP calibrés. 
</p>

<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">SNP ATTENDUS OU NON</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">compte le nombre de SNP par position. hypoth&#232;se : un SNP `calibr&#233;' g&#233;n&#232;re au</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">moins 5 SNP parmi toutes les s&#233;quences. sortie dans une table qui sert de</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">query &#224; la fonction =is_position=</span>
snp %&gt;%
    group_by<span style="color: #3a81c3;">(</span>offset_on_subject<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">qplot(data = ., offset_on_subject, count)</span>
    filter<span style="color: #3a81c3;">(</span>count &gt; 5<span style="color: #3a81c3;">)</span> %&gt;%
    select<span style="color: #3a81c3;">(</span>offset_on_subject<span style="color: #3a81c3;">)</span> %&gt;%
    unlist<span style="color: #3a81c3;">()</span> %&gt;%
    as.vector<span style="color: #3a81c3;">()</span> <span style="color: #4e3163;">-&gt;</span>
    position_table

<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> d&#233;termine si la postion sur la s&#233;quence de r&#233;f&#233;rence est un SNP artificiel</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> ou un autre genre de SNP.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> is_position</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">position</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">table</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> "oui" ou "non"</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel Barreto</span>
<span style="color: #6c3163; font-weight: bold;">is_position</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>position, table<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
    ifelse<span style="color: #6c3163;">(</span>position %<span style="color: #3a81c3; font-weight: bold;">in</span>% table, <span style="color: #2d9574;">'oui'</span>, <span style="color: #2d9574;">'non'</span><span style="color: #6c3163;">)</span>
<span style="color: #3a81c3;">}</span>

snp %&gt;%
    rowwise<span style="color: #3a81c3;">()</span> %&gt;%
    mutate<span style="color: #3a81c3;">(</span>position = is_position<span style="color: #6c3163;">(</span>offset_on_subject, position_table<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
    ungroup<span style="color: #3a81c3;">()</span> <span style="color: #4e3163;">-&gt;</span>
    snp

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">gros r&#233;sultat l&#224;.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">on voit seulement 3 mutations S-&gt;W contre 12 W-&gt;S !</span>
pdf<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/bgc_en_action.pdf"</span>, height = 5.8, width = 8.3<span style="color: #3a81c3;">)</span>
snp %&gt;%
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"non"</span><span style="color: #3a81c3;">)</span> %&gt;%
    plot_snp<span style="color: #3a81c3;">(</span>fill_by = <span style="color: #2d9574;">"mutation_type"</span><span style="color: #3a81c3;">)</span> +
    scale_y_continuous<span style="color: #3a81c3;">(</span>breaks = c<span style="color: #6c3163;">(</span>1, 2<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> +
    scale_x_continuous<span style="color: #3a81c3;">(</span>breaks = position_table, name = waiver<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> +
    ggtitle<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"Substitution aux positions inattendues : biaisees vers GC ?"</span><span style="color: #3a81c3;">)</span> +
    theme<span style="color: #3a81c3;">(</span>panel.grid.major.x = element_line<span style="color: #6c3163;">(</span>colour = <span style="color: #2d9574;">"gray"</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>
dev.off<span style="color: #3a81c3;">()</span>

sink<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/bgc_en_action.tex"</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"non"</span><span style="color: #3a81c3;">)</span> %&gt;%
    group_by<span style="color: #3a81c3;">(</span>mutation_type<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> <span style="color: #4e3163;">-&gt;</span>
    bgc_en_action
colnames<span style="color: #3a81c3;">(</span>bgc_en_action<span style="color: #3a81c3;">)</span> <span style="color: #4e3163;">&lt;-</span> c<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"Type de Substitution"</span>, <span style="color: #2d9574;">"Nombre"</span><span style="color: #3a81c3;">)</span>
print<span style="color: #3a81c3;">(</span>xtable::xtable<span style="color: #6c3163;">(</span> bgc_en_action, align = <span style="color: #2d9574;">'ccc'</span><span style="color: #6c3163;">)</span>, include.rownames = <span style="color: #ba2f59; font-weight: bold;">FALSE</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline22" class="outline-5">
<h5 id="orgheadline22"><span class="section-number-5">3.1.6.6</span> Positions terminales de switch <code>[1/3]</code></h5>
<div class="outline-text-5" id="text-3-1-6-6">
<ul class="org-ul">
<li class="off"><code>[&#xa0;]</code> voir avec vincent le test à utiliser pour comparer les distributions</li>
<li class="off"><code>[&#xa0;]</code> comparer les distributions statistiquement, si \(n\) est suffisamment grand.</li>
<li class="on"><code>[X]</code> Naïvement on utilise ici la position terminale du dernier SNP. Cependant,
il existe des SNPs qui ne sont pas dans la conversion track, c'est ce qu'on a
observé. Il faut donc définir une nouvelle fonction <code>max_pos</code> qui définit si
on est bien à un SNP calibré ou non. voir après avoir définit la fonction
<code>is_a_position</code>. =&gt; Simplement filtrer par position</li>
</ul>

<p>
On veut ici analyser la distribution des positions terminales de SNP. C'est à dire à quel
endroit on bascule à nouveau sur le génotype sauvage. On ne garde que les
positions qui sont des positions calibrées. 
</p>

<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">POSITION DE SWITCH</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>

pdf<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/switch_distrib.pdf"</span>, height = 5.8, width = 8.3<span style="color: #3a81c3;">)</span>

snp %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">par plasmide -- et par type de mutant</span>
    group_by<span style="color: #3a81c3;">(</span>name, mutant<span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">garde seulement les positions calibr&#233;es</span>
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"oui"</span><span style="color: #3a81c3;">)</span> %&gt;% 
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">cherche la postion minimale de SNP</span>
    summarise<span style="color: #3a81c3;">(</span>offset_on_subject = min<span style="color: #6c3163;">(</span>offset_on_subject<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;% 
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">represente la distribution</span>
    plot_snp<span style="color: #3a81c3;">(</span>legend_position = c<span style="color: #6c3163;">(</span>0.8, 0.8<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> +
    ggtitle<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"Distribution de la position de switch en fonction du type de mutant"</span><span style="color: #3a81c3;">)</span> +
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">superpose les deux distribution pour comparer</span>
    facet_grid<span style="color: #3a81c3;">(</span> mutant ~ .<span style="color: #3a81c3;">)</span> 

dev.off<span style="color: #3a81c3;">()</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline23" class="outline-5">
<h5 id="orgheadline23"><span class="section-number-5">3.1.6.7</span> SNP inattendus <code>[1/1]</code></h5>
<div class="outline-text-5" id="text-3-1-6-7">
<ul class="org-ul">
<li class="on"><code>[X]</code> rajouter le filtre sur les positions calibrées.</li>
</ul>

<p>
Le but est de trouver les SNPs aux positions calibrées qui ne sont pas ceux
attendus. Typiquement ce sont les mutations <i>strong</i> dans la manip Weak, et les
mutations <i>weak</i> dans la manip Strong. Si les mutations en questions sont
répétées sur l'ensemble du clone, alors c'est une contamination. Sinon, ce sont
des données intéressantes. 
</p>

<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">============================================================================== </span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">OUTLIERS</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec;">==============================================================================</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">##</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> D&#233;termine si le SNP en question est un outlier ou non, c'est &#224; dire une</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> mutation strong chez un mutant weak ou inversement.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> is_outlier</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">mutant</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> : le type de mutant</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">mutation_type</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> : le type de substitution</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel Barreto</span>
<span style="color: #6c3163; font-weight: bold;">is_outlier</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>mutant, mutation_type<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
    <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #6c3163;">(</span>mutant == <span style="color: #2d9574;">'strong'</span> &amp;&amp; mutation_type == <span style="color: #2d9574;">'SW'</span><span style="color: #6c3163;">)</span> <span style="color: #6c3163;">{</span>
        <span style="color: #2d9574;">"non"</span>
    <span style="color: #6c3163;">}</span> <span style="color: #3a81c3; font-weight: bold;">else</span> <span style="color: #3a81c3; font-weight: bold;">if</span> <span style="color: #6c3163;">(</span>mutant == <span style="color: #2d9574;">'weak'</span> &amp;&amp; mutation_type == <span style="color: #2d9574;">'WS'</span><span style="color: #6c3163;">)</span> <span style="color: #6c3163;">{</span>
        <span style="color: #2d9574;">"non"</span>
    <span style="color: #6c3163;">}</span> <span style="color: #3a81c3; font-weight: bold;">else</span> <span style="color: #6c3163;">{</span>
        <span style="color: #2d9574;">'oui'</span>
    <span style="color: #6c3163;">}</span>
<span style="color: #3a81c3;">}</span>

snp %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">par ligne, determine si la position est inattendue</span>
    rowwise<span style="color: #3a81c3;">()</span> %&gt;%
    mutate<span style="color: #3a81c3;">(</span>attendu = is_outlier<span style="color: #6c3163;">(</span>mutant, mutation_type<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
    ungroup<span style="color: #3a81c3;">()</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">garde les positions calibree</span>
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"oui"</span><span style="color: #3a81c3;">)</span> %&gt;% 
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">sur lesquelles le r&#233;sultat est inattendu</span>
    filter<span style="color: #3a81c3;">(</span>attendu == <span style="color: #2d9574;">"non"</span><span style="color: #3a81c3;">)</span> %&gt;%
    plot_snp<span style="color: #3a81c3;">(</span>legend_position = c<span style="color: #6c3163;">(</span>0.2, 0.9<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> +
    geom_text<span style="color: #3a81c3;">(</span>aes<span style="color: #6c3163;">(</span>label = name, y = 0.5<span style="color: #6c3163;">)</span>, check_overlap = <span style="color: #ba2f59; font-weight: bold;">TRUE</span>,
              position = <span style="color: #2d9574;">"dodge"</span><span style="color: #3a81c3;">)</span> +
    coord_flip<span style="color: #3a81c3;">()</span> +
    theme<span style="color: #3a81c3;">(</span>panel.ontop = <span style="color: #ba2f59; font-weight: bold;">FALSE</span><span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">sortie des r&#233;sultats dans un joli tableau latex</span>
sink<span style="color: #3a81c3;">(</span> file = <span style="color: #2d9574;">"../../analysis/outlier.tex"</span>, append = <span style="color: #ba2f59; font-weight: bold;">FALSE</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    rowwise<span style="color: #3a81c3;">()</span> %&gt;%
    mutate<span style="color: #3a81c3;">(</span>attendu = is_outlier<span style="color: #6c3163;">(</span>mutant, mutation_type<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
    ungroup<span style="color: #3a81c3;">()</span> %&gt;%
    filter<span style="color: #3a81c3;">(</span>attendu == <span style="color: #2d9574;">"non"</span><span style="color: #3a81c3;">)</span> %&gt;%
    mutate<span style="color: #3a81c3;">(</span>position = offset_on_subject, ref = s_base, read = q_base<span style="color: #3a81c3;">)</span> %&gt;%
    select<span style="color: #3a81c3;">(</span>-read_name, -offset_on_subject, -s_base, -q_base<span style="color: #3a81c3;">)</span> %&gt;%
    knitr::kable<span style="color: #3a81c3;">(</span>format = <span style="color: #2d9574;">"latex"</span>, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">pdf(file = "../../analysis/outliers.pdf", width = 4, height = 2)</span>
</pre>
</div>
</div>
</div>
<div id="outline-container-orgheadline24" class="outline-5">
<h5 id="orgheadline24"><span class="section-number-5">3.1.6.8</span> Position des SNP par rapport à la conversion tract <code>[1/1]</code></h5>
<div class="outline-text-5" id="text-3-1-6-8">
<ul class="org-ul">
<li class="on"><code>[X]</code> Écrire une fonction <code>is_inside_conv</code> qui détermine si un SNP est en dehors
ou en dedans de la conversion tract.</li>
</ul>

<div class="org-src-container">

<pre class="src src-R">snp %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">par exogene</span>
    group_by<span style="color: #3a81c3;">(</span>name<span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">garde seulement les positions attendues</span>
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"oui"</span><span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">cherche la borne sup&#233;rieure et inf&#233;rieure</span>
    summarise<span style="color: #3a81c3;">(</span>min = min<span style="color: #6c3163;">(</span>offset_on_subject<span style="color: #6c3163;">)</span>, max = max<span style="color: #6c3163;">(</span>offset_on_subject<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">combine avec la table m&#232;re</span>
    inner_join<span style="color: #3a81c3;">(</span>snp, by = <span style="color: #2d9574;">"name"</span><span style="color: #3a81c3;">)</span> <span style="color: #4e3163;">-&gt;</span>
    snp

<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> d&#233;termine si le SNP est dans la conversion tract ou non. </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">query</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la position requ&#234;te</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">min</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la borne inf&#233;rieure de la conversion tract</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">max</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la borne sup&#233;rieure de la conversion tract</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> oui ou non</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel Barreto</span>
<span style="color: #6c3163; font-weight: bold;">is_inside_conv</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>query, min, max<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
    ifelse<span style="color: #6c3163;">(</span>min &lt;= query &amp; query &lt;= max, <span style="color: #2d9574;">"oui"</span>, <span style="color: #2d9574;">"non"</span><span style="color: #6c3163;">)</span>
<span style="color: #3a81c3;">}</span>

sink<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/inside_conv.tex"</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    rowwise<span style="color: #3a81c3;">()</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">d&#233;termine si on est dans la conversion tract ou non</span>
    mutate<span style="color: #3a81c3;">(</span>inside_conv = is_inside_conv<span style="color: #6c3163;">(</span>offset_on_subject, min, max<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
    ungroup<span style="color: #3a81c3;">()</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">filtre pour avoir les snp non attendus</span>
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"non"</span><span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">groupe selon qu'on est dans ou en dehors de la CT</span>
    group_by<span style="color: #3a81c3;">(</span>inside_conv<span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">compte le nombre de snp par cas</span>
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">format en .tex</span>
    knitr::kable<span style="color: #3a81c3;">(</span>format = <span style="color: #2d9574;">"latex"</span>, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline27" class="outline-5">
<h5 id="orgheadline27"><span class="section-number-5">3.1.6.9</span> Polymorphisme <code>[1/2]</code></h5>
<div class="outline-text-5" id="text-3-1-6-9">
</div><ol class="org-ol"><li><a id="orgheadline25"></a>Graphiquement<br  /><div class="outline-text-6" id="text-3-1-6-9-1">
<p>
On veux étudier le polymorphisme aux positions calibrées. Comme on ne peut pas
avoir accès aux données trace par trace, on utilise la qualité  comme proxy du
polymorphisme. Résultat intéressant ici. 
</p>

<div class="org-src-container">

<pre class="src src-R">pdf<span style="color: #3a81c3;">(</span>file = <span style="color: #2d9574;">"../../analysis/qualite_distrib.pdf"</span>, height = 5.8, width = 8.3<span style="color: #3a81c3;">)</span>
snp %&gt;%
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"oui"</span><span style="color: #3a81c3;">)</span> %&gt;%
    qplot<span style="color: #3a81c3;">(</span>data = ., x = offset_on_subject, y = q_qual, color = mutant,
          alpha = 0.1<span style="color: #3a81c3;">)</span> +
    theme_minimal<span style="color: #3a81c3;">(</span>base_family = <span style="color: #2d9574;">"Courier"</span><span style="color: #3a81c3;">)</span> +
    scale_color_brewer<span style="color: #3a81c3;">(</span>palette = <span style="color: #2d9574;">"Set2"</span>, name = <span style="color: #2d9574;">""</span><span style="color: #3a81c3;">)</span> +
    scale_alpha_continuous<span style="color: #3a81c3;">(</span>guide = <span style="color: #ba2f59; font-weight: bold;">FALSE</span>, name = <span style="color: #2d9574;">""</span><span style="color: #3a81c3;">)</span> + 
    xlab<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"Position"</span><span style="color: #3a81c3;">)</span> + ylab<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"qualite"</span><span style="color: #3a81c3;">)</span> +
    scale_x_continuous<span style="color: #3a81c3;">(</span>breaks = position_table<span style="color: #3a81c3;">)</span> +
    theme<span style="color: #3a81c3;">(</span>legend.position = c<span style="color: #6c3163;">(</span>0.5, 0.2<span style="color: #6c3163;">)</span>,
          axis.text = element_text<span style="color: #6c3163;">(</span>size = 8, colour = <span style="color: #2d9574;">"gray"</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>
dev.off<span style="color: #3a81c3;">()</span>
</pre>
</div>

<p>
<a href="./analysis/qualite_distrib.pdf">./analysis/qualite_distrib.pdf</a>
</p>
</div></li>
<li><a id="orgheadline26"></a>Fréquence des bases polymorphes associées aux SNP attendus<br  /><div class="outline-text-6" id="text-3-1-6-9-2">
<div class="org-src-container">

<pre class="src src-R">sink<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"../../analysis/freq_polymorphism.tex"</span><span style="color: #3a81c3;">)</span>
snp %&gt;%
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"oui"</span>, q_qual &lt; 40, offset_on_subject &lt; 600<span style="color: #3a81c3;">)</span> %&gt;%
    group_by<span style="color: #3a81c3;">(</span>mutant<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span> %&gt;%
    knitr::kable<span style="color: #3a81c3;">(</span>format = <span style="color: #2d9574;">"latex"</span>, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #3a81c3;">)</span>
sink<span style="color: #3a81c3;">()</span>

snp %&gt;%
    filter<span style="color: #3a81c3;">(</span>position == <span style="color: #2d9574;">"non"</span>, q_qual &lt; 40, offset_on_subject &lt; 600<span style="color: #3a81c3;">)</span> <span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">%&gt;%</span>
    group_by<span style="color: #3a81c3;">(</span>mutant<span style="color: #3a81c3;">)</span> %&gt;%
    summarise<span style="color: #3a81c3;">(</span>count = n<span style="color: #6c3163;">()</span><span style="color: #3a81c3;">)</span><span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">%&gt;%</span>
    knitr::kable<span style="color: #3a81c3;">(</span>format = <span style="color: #2d9574;">"latex"</span>, booktabs = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #3a81c3;">)</span>
</pre>
</div>
</div></li></ol>
</div>
</div>
</div>
<div id="outline-container-orgheadline35" class="outline-3">
<h3 id="orgheadline35"><span class="section-number-3">3.2</span> Deuxième version</h3>
<div class="outline-text-3" id="text-3-2">
</div><div id="outline-container-orgheadline30" class="outline-4">
<h4 id="orgheadline30"><span class="section-number-4">3.2.1</span> Base calling</h4>
<div class="outline-text-4" id="text-3-2-1">
<div class="org-src-container">

<pre class="src src-R">setwd<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"~/stage/seq_novembre/data/tmp/"</span><span style="color: #3a81c3;">)</span>
<span style="color: #4e3163;">library</span><span style="color: #3a81c3;">(</span>dplyr<span style="color: #3a81c3;">)</span>
<span style="color: #4e3163;">library</span><span style="color: #3a81c3;">(</span>sangerseqR<span style="color: #3a81c3;">)</span>
<span style="color: #4e3163;">library</span><span style="color: #3a81c3;">(</span>Biostrings<span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">la deuxi&#232;me ligne du fichier reference.fasta contient la s&#233;quence de</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">r&#233;f&#233;rence. Elle est convertie en un objet DNA string.</span>
ref <span style="color: #4e3163;">&lt;-</span> readLines<span style="color: #3a81c3;">(</span><span style="color: #2d9574;">"../reference.fasta"</span><span style="color: #3a81c3;">)[</span>2<span style="color: #3a81c3;">]</span> %&gt;%
  DNAString<span style="color: #3a81c3;">()</span>

<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Call bases on a query, based on primary and secondary traces.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> baseCaller</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">query</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> the ab1 file name.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">ref</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> the reference. a DNAString object.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> a sangerseqR object.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel BARRETO</span>
<span style="color: #6c3163; font-weight: bold;">baseCaller</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>query, ref<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
  readsangerseq<span style="color: #6c3163;">(</span>query<span style="color: #6c3163;">)</span> %&gt;%
    makeBaseCalls<span style="color: #6c3163;">(</span>ratio = 0.5<span style="color: #6c3163;">)</span><span style="color: #2aa1ae; background-color: #ecf3ec;">#  </span><span style="color: #2aa1ae; background-color: #ecf3ec;">%&gt;%</span>
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">setAllelePhase(obj = ., refseq = ref)</span>
<span style="color: #3a81c3;">}</span>

<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> TEST: pW85 &lt;- baseCaller(query = "../spectrograms/pW85-1073.ab1", ref = ref)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline31" class="outline-4">
<h4 id="orgheadline31"><span class="section-number-4">3.2.2</span> Alignement</h4>
<div class="outline-text-4" id="text-3-2-2">
<div class="org-src-container">

<pre class="src src-R"><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Aligne les donn&#233;es de la s&#233;quence primaire avec ceux de la s&#233;quence</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> secondaire, en utilisant un algorithme TODO de type SM.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> aligner</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">seq</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> l'objet sangerseq d'int&#233;r&#234;t.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">name</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> le nom de la s&#233;quence, NSE.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> un tableau contenant la position, la base du pic majeur,</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> la base du pic secondaire et le nom de la s&#233;quence.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel BARRETO</span>
<span style="color: #6c3163; font-weight: bold;">aligner</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>seq, name<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
  argname <span style="color: #4e3163;">&lt;-</span> deparse<span style="color: #6c3163;">(</span>substitute<span style="color: #2d9574;">(</span>seq<span style="color: #2d9574;">)</span><span style="color: #6c3163;">)</span>
  pairwiseAlignment<span style="color: #6c3163;">(</span>primarySeq<span style="color: #2d9574;">(</span>seq<span style="color: #2d9574;">)</span>, secondarySeq<span style="color: #2d9574;">(</span>seq<span style="color: #2d9574;">)</span>,
                    type= <span style="color: #2d9574;">"global-local"</span><span style="color: #6c3163;">)</span> %&gt;%
    mismatchTable<span style="color: #6c3163;">()</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">ne garde que les bases litigieuses et leur position</span>
    select<span style="color: #6c3163;">(</span>primaire   = PatternSubstring,
           secondaire = SubjectSubstring,
           position   = PatternStart<span style="color: #6c3163;">)</span> %&gt;%
    <span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">trim les positions de faible qualit&#233;</span>
    filter<span style="color: #6c3163;">(</span>position &gt; 40, <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">premi&#232;re position de bonne qualit&#233;</span>
           position &lt; length<span style="color: #2d9574;">(</span>primarySeq<span style="color: #67b11d;">(</span>seq<span style="color: #67b11d;">)</span><span style="color: #2d9574;">)</span> - 40<span style="color: #6c3163;">)</span> %&gt;% <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">derni&#232;re position de bonne qualit&#233;.</span>
    mutate<span style="color: #6c3163;">(</span>name = name<span style="color: #6c3163;">)</span> <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">ajoute une colonne avec le nom de la s&#233;quence.</span>
<span style="color: #3a81c3;">}</span>

pW85_mm <span style="color: #4e3163;">&lt;-</span> aligner<span style="color: #3a81c3;">(</span>pW85, <span style="color: #2d9574;">"pW85"</span><span style="color: #3a81c3;">)</span>
pW85_mm
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline32" class="outline-4">
<h4 id="orgheadline32"><span class="section-number-4">3.2.3</span> Pooling</h4>
<div class="outline-text-4" id="text-3-2-3">
<div class="org-src-container">

<pre class="src src-R">id_table <span style="color: #4e3163;">&lt;-</span> tbl_df<span style="color: #3a81c3;">(</span>read.table<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"../id_table.dat"</span>, stringsAsFactors = <span style="color: #ba2f59; font-weight: bold;">FALSE</span>, head = <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">rebascule les donn&#233;es litigieuses dans les bonnes cat&#233;gories</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">voir les premiers scripts.</span>
id_table$mutant<span style="color: #3a81c3;">[</span>id_table$name == <span style="color: #2d9574;">"pS60-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
id_table$mutant<span style="color: #3a81c3;">[</span>id_table$name == <span style="color: #2d9574;">"pS83-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
id_table$mutant<span style="color: #3a81c3;">[</span>id_table$name == <span style="color: #2d9574;">"pS92-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
id_table$mutant<span style="color: #3a81c3;">[</span>id_table$name == <span style="color: #2d9574;">"pS91-1073"</span><span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"weak"</span>
id_table$mutant<span style="color: #3a81c3;">[</span>id_table$name == <span style="color: #2d9574;">"pW6-1073"</span> <span style="color: #3a81c3;">]</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #2d9574;">"strong"</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">&#233;limine les fichiers aux donn&#233;es manquantes</span>
id_table <span style="color: #4e3163;">&lt;-</span> filter<span style="color: #3a81c3;">(</span>id_table, name != <span style="color: #2d9574;">"pS6-1073"</span>, name != <span style="color: #2d9574;">"pS9-1073"</span><span style="color: #3a81c3;">)</span>

<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \description{} (no empty lines) ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Appelle les bases principales et secondaires par fichier,</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> et aligne avec la s&#233;quence de r&#233;f&#233;rence.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> .. content for \details{} ..</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@title</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> mismatcher</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">sequence</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> char, le nom de la s&#233;quence.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@param</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #715ab1; background-color: #ecf3ec;">ref</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> DNAString, la s&#233;quence de r&#233;f&#233;rence.</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@return</span>
<span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">##</span><span style="color: #2aa1ae; background-color: #ecf3ec; font-weight: bold;">'</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> </span><span style="color: #3a81c3; background-color: #ecf3ec; font-weight: bold;">@author</span><span style="color: #2aa1ae; background-color: #ecf3ec;"> Samuel BARRETO</span>
<span style="color: #6c3163; font-weight: bold;">mismatcher</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>sequence, ref<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>
  baseCaller<span style="color: #6c3163;">(</span>query = paste0<span style="color: #2d9574;">(</span><span style="color: #2d9574;">"../spectrograms/"</span>, sequence, <span style="color: #2d9574;">".ab1"</span><span style="color: #2d9574;">)</span>, ref<span style="color: #6c3163;">)</span> %&gt;%
  aligner<span style="color: #6c3163;">(</span>., sequence<span style="color: #6c3163;">)</span>
<span style="color: #3a81c3;">}</span>


<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">baseCaller(query = paste0("../spectrograms/", id_table$name[1], ".ab1"), ref)</span>
<span style="color: #2aa1ae; background-color: #ecf3ec;">## </span><span style="color: #2aa1ae; background-color: #ecf3ec;">test_id_table &lt;- id_table %&gt;% sample_n(30)</span>

result_table <span style="color: #4e3163;">&lt;-</span> do.call<span style="color: #3a81c3;">(</span>
  rbind,
  lapply<span style="color: #6c3163;">(</span>
    lapply<span style="color: #2d9574;">(</span>id_table$name, <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #67b11d;">(</span>x<span style="color: #67b11d;">)</span> mismatcher<span style="color: #67b11d;">(</span>sequence = x, ref = ref<span style="color: #67b11d;">)</span><span style="color: #2d9574;">)</span>,
    data.frame
  <span style="color: #6c3163;">)</span>
<span style="color: #3a81c3;">)</span>

write.csv<span style="color: #3a81c3;">(</span>result_table, file = <span style="color: #2d9574;">"result_table"</span><span style="color: #3a81c3;">)</span>

result_table %&gt;%
  tbl_df<span style="color: #3a81c3;">()</span> %&gt;%
  inner_join<span style="color: #3a81c3;">(</span>id_table, by = <span style="color: #2d9574;">"name"</span><span style="color: #3a81c3;">)</span> %&gt;%
  filter<span style="color: #3a81c3;">(</span>mutant == <span style="color: #2d9574;">"strong"</span><span style="color: #3a81c3;">)</span> %&gt;%
  print<span style="color: #3a81c3;">()</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline33" class="outline-4">
<h4 id="orgheadline33"><span class="section-number-4">3.2.4</span> Sens des mutations</h4>
<div class="outline-text-4" id="text-3-2-4">
<div class="org-src-container">

<pre class="src src-R">iupac_code <span style="color: #4e3163;">&lt;-</span>
<span style="color: #2d9574;">"</span>
<span style="color: #2d9574;">iupac,meaning</span>
<span style="color: #2d9574;">A,A</span>
<span style="color: #2d9574;">C,C</span>
<span style="color: #2d9574;">G,G</span>
<span style="color: #2d9574;">T,T</span>
<span style="color: #2d9574;">M,AC</span>
<span style="color: #2d9574;">R,AG</span>
<span style="color: #2d9574;">W,AT</span>
<span style="color: #2d9574;">S,CG</span>
<span style="color: #2d9574;">Y,CT</span>
<span style="color: #2d9574;">K,GT</span>
<span style="color: #2d9574;">V,ACG</span>
<span style="color: #2d9574;">H,ACT</span>
<span style="color: #2d9574;">D,AGT</span>
<span style="color: #2d9574;">B,CGT</span>
<span style="color: #2d9574;">X,GATC</span>
<span style="color: #2d9574;">N,GATC</span>
<span style="color: #2d9574;">"</span>
read.csv<span style="color: #3a81c3;">(</span>text = iupac_code, stringsAsFactors = <span style="color: #ba2f59; font-weight: bold;">FALSE</span><span style="color: #3a81c3;">)</span> %&gt;% tbl_df<span style="color: #3a81c3;">()</span> <span style="color: #4e3163;">-&gt;</span> iupac

<span style="color: #6c3163; font-weight: bold;">is_iupac</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>base<span style="color: #3a81c3;">)</span> ifelse<span style="color: #3a81c3;">(</span>base %<span style="color: #3a81c3; font-weight: bold;">in</span>% c<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"A"</span>, <span style="color: #2d9574;">"T"</span>, <span style="color: #2d9574;">"C"</span>, <span style="color: #2d9574;">"G"</span><span style="color: #6c3163;">)</span>, <span style="color: #ba2f59; font-weight: bold;">FALSE</span>, <span style="color: #ba2f59; font-weight: bold;">TRUE</span><span style="color: #3a81c3;">)</span>

<span style="color: #6c3163; font-weight: bold;">iupacker</span> <span style="color: #4e3163;">&lt;-</span> <span style="color: #3a81c3; font-weight: bold;">function</span><span style="color: #3a81c3;">(</span>subject, query<span style="color: #3a81c3;">)</span>
<span style="color: #3a81c3;">{</span>

  lapply<span style="color: #6c3163;">(</span>
    str_split<span style="color: #2d9574;">(</span>paste0<span style="color: #67b11d;">(</span>subject, query<span style="color: #67b11d;">)</span>, <span style="color: #2d9574;">""</span><span style="color: #2d9574;">)</span>,
    sort<span style="color: #6c3163;">)</span>
<span style="color: #3a81c3;">}</span>

result_table %&gt;%
  filter<span style="color: #3a81c3;">(</span> is_iupac<span style="color: #6c3163;">(</span>SubjectSubstring<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
  inner_join<span style="color: #3a81c3;">(</span>y = iupac, by = c<span style="color: #6c3163;">(</span><span style="color: #2d9574;">"SubjectSubstring"</span> = <span style="color: #2d9574;">"iupac"</span><span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span>
</pre>
</div>
</div>
</div>

<div id="outline-container-orgheadline34" class="outline-4">
<h4 id="orgheadline34"><span class="section-number-4">3.2.5</span> Analyses</h4>
<div class="outline-text-4" id="text-3-2-5">
<div class="org-src-container">

<pre class="src src-R">peakAmpMatrix<span style="color: #3a81c3;">(</span>pW85<span style="color: #3a81c3;">)</span> %&gt;%
  as.data.frame<span style="color: #3a81c3;">()</span> %&gt;%
  tbl_df<span style="color: #3a81c3;">()</span> %&gt;%
  mutate<span style="color: #3a81c3;">(</span>pos = rownames<span style="color: #6c3163;">(</span>.<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
  gather<span style="color: #3a81c3;">(</span>trace, value, V1:V4<span style="color: #3a81c3;">)</span> <span style="color: #4e3163;">-&gt;</span> peakmatrix

traceMatrix<span style="color: #3a81c3;">(</span>pW85<span style="color: #3a81c3;">)</span> %&gt;%
  as.data.frame<span style="color: #3a81c3;">()</span> <span style="color: #2aa1ae; background-color: #ecf3ec;">#</span><span style="color: #2aa1ae; background-color: #ecf3ec;">%&gt;%</span>
  tbl_df<span style="color: #3a81c3;">()</span> %&gt;%
  mutate<span style="color: #3a81c3;">(</span>pos = rownames<span style="color: #6c3163;">(</span>.<span style="color: #6c3163;">)</span><span style="color: #3a81c3;">)</span> %&gt;%
  gather<span style="color: #3a81c3;">(</span>trace, value, V1:V4<span style="color: #3a81c3;">)</span> <span style="color: #4e3163;">-&gt;</span> tracematrix
</pre>
</div>
</div>
</div>
</div>
</div>

<div id="outline-container-orgheadline37" class="outline-2">
<h2 id="orgheadline37"><span class="section-number-2">4</span> Script Principal</h2>
<div class="outline-text-2" id="text-4">
<div class="org-src-container">

<pre class="src src-sh"><span style="color: #3a81c3;">cd</span> ~/stage/seq_novembre/scripts

chmod +x *.sh

./extract_raw_data.sh                          <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">extrait les donn&#233;es des .zip et</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">organise en sous fichiers</span>

./ab1_to_fastq.sh                              <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">extrait les fastq et fasta</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">depuis les .ab1</span>

./trim_low_quality.sh                          <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">supprime les donn&#233;es de faible</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">qualit&#233; et transforme les bases</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">de qualit&#233; inf&#233;rieure &#224; 28 en N</span>

./quality_check.sh                             <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">analyse fastqc, cr&#233;e les</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">graphes en png et sortie des</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">html dans le dossier analysis</span>

./variantCallerSsaha2.sh                       <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">d&#233;termine la position des SNP</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">bas&#233; sur la r&#233;f&#233;rence</span>

python make_id_table.py &gt; ../data/id_table.dat <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">cr&#233;e la table d'association des</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">identifiants de s&#233;quence avec</span>
                                               <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">le nom des clones</span>

Rscript variant_analysis.R                     <span style="color: #2aa1ae; background-color: #ecf3ec;"># </span><span style="color: #2aa1ae; background-color: #ecf3ec;">analyses et graphiques</span>
</pre>
</div>
</div>
</div>


<div id="outline-container-orgheadline38" class="outline-2">
<h2 id="orgheadline38"><span class="section-number-2">5</span> configuration&#xa0;&#xa0;&#xa0;<span class="tag"><span class="CONFIG">CONFIG</span></span></h2>
<div class="outline-text-2" id="text-5">
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="author">Author: Samuel BARRETO</p>
<p class="date">Created: 2016-01-04 Mon 16:56</p>
<p class="validation"><a href="http://validator.w3.org/check?uri=referer">Validate</a></p>
</div>
</body>
</html>
